### ISSUE TITLE
why test batch size will affect my acc?

### ISSUE THREAD

**Issue by Wenboisstrong** [xuexingyu24/License_Plate_Detection_Pytorch#22]
The batch size of my training model is 128. When I test the batch size is also 128. The model acc is 95%. When I set the test batch size to 1, the model acc drops a lot. Only 80%. Why is there such a result. I confirm that I use Model.eval () enters evaluation mode

**Comment by charlestamz** [comment]
try something like :
https://discuss.pytorch.org/t/how-to-absorb-batch-norm-layer-weights-into-convolution-layer-weights/16412

**Comment by Wenboisstrong** [comment]
@charlestamz  Thank you for your suggestion. I successfully integrated the BN layer into the conv and I got the same result as before. So I think that if the model is set.Eval ()    BN layer and dropout layer will not affect the result. May appear in the loading data or calculation results I will continue to track this problem

**Comment by Wenboisstrong** [comment]
It seems that most of the stars are not particularly interested in this project. Finally, I found the problem https://github.com/xuexingyu24/License_Plate_Detection_Pytorch/blob/aa5778d6e6da0f382894048b4199cca7a1151e95/LPRNet/model/LPRNET.py#L78
 The problem is () when f_pow is a batch of data. torch. mean () will get a mean of btach, but torch when f_pow is a single data. mean
  Only ask for the mean of a single piece of data. I think the problem is the key here and the model has been modified according to the official code. In the case of using the same data, acc is from 95%-> 97%

**Comment by Wenboisstrong** [comment]
You can refer to my blog  https://blog.csdn.net/weixin_39027619/article/details/106143755
@loveandhope

**Comment by loveandhope** [comment]
THANK YOU VERY MUCH! @Wenboisstrong

**Comment by Aslyamovt** [comment]
Thank you very much! @Wenboisstrong 

We found the way to fix the problem
You have to replace this [https://github.com/xuexingyu24/License_Plate_Detection_Pytorch/blob/aa5778d6e6da0f382894048b4199cca7a1151e95/LPRNet/model/LPRNET.py#L78](url) 
to this code 
`f_mean = torch.mean(torch.mean(torch.mean(f_pow,axis = -1),axis = -1),axis = -1)
 f_mean = torch.unsqueeze(f_mean, 1)
 f_mean = torch.unsqueeze(f_mean, 1)
 f_mean = torch.unsqueeze(f_mean, 1)`
and anything will work fine!
